.evergreen/run-typescript.sh

```diff
@@ -44,8 +44,12 @@ if [ "$TS_CHECK" == "COMPILE_DRIVER" ]; then
     npm run build:ts
 elif [ "$TS_CHECK" == "CHECK_TYPES" ]; then
     echo "checking driver types"
+    if [ "$TS_VERSION" == "4.4" ]; then
     # check compilation
-    node $TSC mongodb.d.ts
+        node $TSC mongodb.d.ts --module commonjs --target es2021
```

Comment:
Are we forced to set a module when specifying target?

Reply:
Yup.  By default the target is es5, which has no built-in module system so TS doesn't need a module resolution system.  Anything es6 or later must have the `-m` option specified